src/dbus/server/dbus_thread_object.cpp

```diff
@@ -1254,10 +1254,20 @@ otError DBusThreadObject::GetSrpServerInfoHandler(DBusMessageIter &aIter)
 otError DBusThreadObject::GetDnssdCountersHandler(DBusMessageIter &aIter)
 {
 #if OTBR_ENABLE_DNSSD_DISCOVERY_PROXY
-    auto          threadHelper  = mNcp->GetThreadHelper();
-    auto          instance      = threadHelper->GetInstance();
-    otError       error         = OT_ERROR_NONE;
-    DnssdCounters dnssdCounters = *otDnssdGetCounters(instance);
+    auto            threadHelper = mNcp->GetThreadHelper();
+    auto            instance     = threadHelper->GetInstance();
+    otError         error        = OT_ERROR_NONE;
+    DnssdCounters   dnssdCounters;
+    otDnssdCounters otDnssdCounters = *otDnssdGetCounters(instance);
+
+    dnssdCounters.mSuccessResponse        = otDnssdCounters.mSuccessResponse;
+    dnssdCounters.mServerFailureResponse  = otDnssdCounters.mServerFailureResponse;
+    dnssdCounters.mFormatErrorResponse    = otDnssdCounters.mFormatErrorResponse;
+    dnssdCounters.mNameErrorResponse      = otDnssdCounters.mNameErrorResponse;
+    dnssdCounters.mNotImplementedResponse = otDnssdCounters.mNotImplementedResponse;
+    dnssdCounters.mOtherResponse          = otDnssdCounters.mOtherResponse;
+
+    dnssdCounters.mResolvedBySrp = otDnssdCounters.mResolvedBySrp;
 
     VerifyOrExit(DBusMessageEncodeToVariant(&aIter, dnssdCounters) == OTBR_ERROR_NONE, error = OT_ERROR_INVALID_ARGS);
```

Comment:
I think you'll need to define the corresponding `DBusMessageEncodeToVariant` implementation for `DnssdCounters`.